c++/src/capnp/any.h

```diff
@@ -806,6 +806,31 @@ inline Orphan<AnyPointer> Orphan<AnyPointer>::releaseAs() {
   return kj::mv(*this);
 }
 
+enum class StructEqualityResult {
+  NOT_EQUAL,
+  EQUAL,
+  UNKNOWN_CONTAINS_CAPS
+};
+
+inline kj::StringPtr KJ_STRINGIFY(StructEqualityResult res) {
+  switch(res) {
+  case StructEqualityResult::NOT_EQUAL:
+    return "NOT_EQUAL";
+  case StructEqualityResult::EQUAL:
+    return "EQUAL";
+  case StructEqualityResult::UNKNOWN_CONTAINS_CAPS:
+    return "UNKNOWN_CONTAINS_CAPS";
+  }
+}
+
+StructEqualityResult equal(AnyStruct::Reader left, AnyStruct::Reader right);
```

Comment:
That's fine.  I added it to a Any{Struct,List,Pointer}::{Reader,Builder} for symmetry.  That leads to some duplication that I'm not sure how to get rid of (`operator ==`).